cmd/sidecars/qemu-args/main_test.go

```diff
@@ -0,0 +1,111 @@
+package main
+
+import (
+	"log"
+	"os"
+	"reflect"
+	"sort"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+var _ = Describe("with custom QEMU command line options", func() {
+	Context("Merging custom QEMU command line options", func() {
+		logger = log.New(os.Stderr, "test", log.Ldate)
+
+		DescribeTable("with different configurations", func(curArgs, curEnvs, annArgs, annEnvs, resArgs, resEnvs []string) {
+			var annotation = createCommandLine(annArgs, annEnvs)
+			var current = createCommandLine(curArgs, curEnvs)
+			var result = createCommandLine(resArgs, resEnvs)
+
+			var merged = mergeQEMUCmd(current, annotation)
+
+			Expect(merged.QEMUArg).To(HaveLen(len(result.QEMUArg)))
+			Expect(merged.QEMUEnv).To(HaveLen(len(result.QEMUEnv)))
+			// With args, we kept the order intact as that might be important
+			Expect(reflect.DeepEqual(merged.QEMUArg, result.QEMUArg)).To(BeTrue())
```

Comment:
nit: I think this can be simplified. Gomega already uses `reflect.DeepEqual` internally for checking equality https://onsi.github.io/gomega/#equalexpected-interface
```suggestion
			Expect(merged.QEMUArg).To(Equal(result.QEMUArg)
```

Reply:
Yep. Didn't know that. Fixed!